src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -454,6 +454,14 @@ def _check_strategy_and_fallback(self) -> None:
         # TODO this logic should apply to both str and object config
         strategy_flag = "" if isinstance(self._strategy_flag, Strategy) else self._strategy_flag
 
+        if _habana_available_and_importable():
+            from lightning_habana import HPUAccelerator
+
+            if isinstance(self._accelerator_flag, HPUAccelerator):
+                if strategy_flag:
+                    self._strategy_flag = strategy_flag
+                return
```

Comment:
I suggest we respect the return value below, instead of repeating the logic here. It would be easier if we made a function

```py
def _is_hpu_accelerator(accelerator: Accelerator) -> bool:
    if _habana_available_and_importable():
            from lightning_habana import HPUAccelerator

            if isinstance(self._accelerator_flag, HPUAccelerator):
				return True
	return False
```

and adding a call to this as a condition to the FSDP check below.

Reply:
Incorporated the suggestions.